apps/shipments/signals.py

```diff
@@ -117,6 +117,14 @@ def trackingdata_post_save(sender, **kwargs):
                                             {"type": "tracking_data.save", "tracking_data_id": instance.id})
 
 
+@receiver(post_save_changed, sender=Shipment, fields=['delivery_act'], dispatch_uid='shipment_delivery_act_post_save')
+def shipment_delivery_act_changed(sender, instance, changed_fields, **kwargs):
+    logging.info(f'Shipment with id {instance.id} ended on {instance.delivery_act}.')
+
+    instance.device = None
+    instance.save()
```

Comment:
This will trigger the `shipment_post_save` signal, which adds an additional RPC call to add_shipment_data (even though no shipment data is changed) and an update_vault_hash call. We can avoid this by doing:
`device_id = instance.device_id`
`Shipment.objects.filter(id=instance.id).update(device_id=None)`
and then calling
`shipment_device_id_changed(Shipment, instance, {Shipment.device.field: (device_id, None)})`

This approach skips the shipment_post_save signal and then manually fires the shipment_device_id_changed signal.